notifier/notifier.go

```diff
@@ -640,6 +641,17 @@ func newAlertmanagerSet(cfg *config.AlertmanagerConfig, logger log.Logger, metri
 	if err != nil {
 		return nil, err
 	}
+	t := client.Transport
+
+	if cfg.SigV4Config != nil {
+		t, err = sigv4.NewSigV4RoundTripper(cfg.SigV4Config, client.Transport)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	client.Transport = t
```

Comment:
The remote write client also injects a span:
https://github.com/prometheus/prometheus/blob/db816ce9dbf2768ed2db1619fb2a8b9f9addeae8/storage/remote/client.go#L159

Is that also desirable here?

Reply:
resolving since we're limiting to sigv4 here, and this would require adding a `Headers` section to the config as well